packages/diagrams/frontend/sirius-components-diagrams/src/renderer/palette/Palette.types.ts

```diff
@@ -16,6 +16,21 @@ export interface ContextualPaletteStyleProps {
   toolCount: number;
 }
 
+export interface GQLTool {
+  id: string;
+  label: string;
+  iconURL: string[];
+  __typename: string;
+}
+
+export interface GQLToolVariable {
+  name: string;
+  value: string;
+  type: GQLToolVariableType;
+}
+
+export type GQLToolVariableType = 'STRING' | 'OBJECT_ID' | 'OBJECT_ID_ARRAY';
```

Comment:
I don't think we need to maintain three copies of this code

Reply:
Sometimes we have duplicated types through different components. It this case it might not be necessary, I don't know